spec/models/course_user_spec.rb

```diff
@@ -109,6 +109,37 @@
       end
     end
 
+    describe '.ordered_by_experience_points' do
+      let!(:student1) { create(:course_student, course: course) }
+      before { create(:course_experience_points_record, course_user: student) }
+
+      it 'returns course_users sorted by experience points' do
+        expect(course.course_users.student.ordered_by_experience_points).to eq([student, student1])
```

Comment:
alternative way of testing is to create both, then check that each_cons(2) has a.experience_points <= b.experience_points